dGame/dComponents/TriggerComponent.cpp

```diff
@@ -242,51 +249,53 @@ void TriggerComponent::HandlePushObject(Entity* targetEntity, std::vector<std::s
 
 void TriggerComponent::HandleRepelObject(Entity* targetEntity, std::string args){
 	auto* phantomPhysicsComponent = m_Parent->GetComponent<PhantomPhysicsComponent>();
-	if (!phantomPhysicsComponent) return;
+	if (!phantomPhysicsComponent) {
+		Game::logger->Log("TriggerComponent::HandleRepelObject", "Phantom Physics component not found!");
+		return;
+	}
 	float forceMultiplier;
 	GeneralUtils::TryParse<float>(args, forceMultiplier);
 	phantomPhysicsComponent->SetPhysicsEffectActive(true);
-	phantomPhysicsComponent->SetEffectType(1); // repel/repulse
+	phantomPhysicsComponent->SetEffectType(ePhysicsEffectType::REPULSE);
 	phantomPhysicsComponent->SetDirectionalMultiplier(forceMultiplier);
 
 	auto triggerPos = m_Parent->GetPosition();
 	auto targetPos = targetEntity->GetPosition();
 
 	// normalize the vectors to get the direction
 	auto delta = targetPos - triggerPos;
-	auto abs = sqrtf(delta.x * delta.x + delta.y * delta.y + delta.z * delta.z);
-	NiPoint3 direction = NiPoint3::ZERO;
-	direction.x = delta.x / abs;
-	direction.y = delta.y / abs;
-	direction.z = delta.z / abs;
-
+	auto abs = delta.Length();
```

Comment:
Didn't catch this initially, but this variable name overloads a function with the same name <https://cplusplus.com/reference/cstdlib/abs/>
We could be change this so that if abs is used here in the future, there isn't a collision